File: app/src/main/java/com/app/washmania/SplashActivity.java
package com.app.washmania;

import android.Manifest;
import android.content.Context;
import android.content.DialogInterface;
import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.provider.Settings;
import android.support.v7.app.AlertDialog;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;
import com.app.washmania.others.WMPreference;
import com.karumi.dexter.Dexter;
import com.karumi.dexter.PermissionToken;
import com.karumi.dexter.listener.PermissionDeniedResponse;
import com.karumi.dexter.listener.PermissionGrantedResponse;
import com.karumi.dexter.listener.PermissionRequest;
import com.karumi.dexter.listener.single.PermissionListener;

import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.Date;
import java.util.Random;
import java.util.TimeZone;

public class SplashActivity extends AppCompatActivity {

    boolean is_verfied = false;
    Context mContext;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);

        mContext = this;
        if (!WMPreference.INSTANCE.isgenerateUniqueKey(mContext)) {
            generateUniqueId();
            WMPreference.INSTANCE.setUniqueKey(mContext, true);
        }

        is_verfied = WMPreference.INSTANCE.getisVerified(mContext);
        if (WMPreference.INSTANCE.get_userId(mContext).equalsIgnoreCase("0")) {
            WMPreference.INSTANCE.set_userId(mContext, "0");
        }


        Dexter.withActivity(this)
                .withPermission(Manifest.permission.ACCESS_FINE_LOCATION)
                .withListener(new PermissionListener() {
                    @Override
                    public void onPermissionGranted(PermissionGrantedResponse response) {
                        if (is_verfied) {
                            goToHomeActivity();
                        } else {
                            goToRegistrationActivity();
                        }
                    }

                    @Override
                    public void onPermissionDenied(PermissionDeniedResponse response) {
                        if (response.isPermanentlyDenied()) {
                            // open device settings when the permission is
                            // denied permanently
                            openSettings();
                        }
                    }

                    @Override
                    public void onPermissionRationaleShouldBeShown(PermissionRequest permission, PermissionToken token) {
                        token.continuePermissionRequest();
                    }
                }).check();

        // goToHomeActivity();

    }

    private void openSettings() {
        Intent intent = new Intent();
        intent.setAction(
                Settings.ACTION_APPLICATION_DETAILS_SETTINGS);
        Uri uri = Uri.fromParts("package",
                BuildConfig.APPLICATION_ID, null);
        intent.setData(uri);
        intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
        startActivity(intent);
    }



    // navigating user to app settings
    private void goToHomeActivity() {
        Thread background = new Thread() {
            @Override
            public void run() {
                try {
                    sleep(3 * 1000);
                    Intent i = new Intent(getBaseContext(), DashboardActivity.class);
                    startActivity(i);
                    finish();
                } catch (Exception e) {
                }
            }
        };
        background.start();
    }

    private void generateUniqueId() {
        Random rand = new Random();
        String uniqueId = System.currentTimeMillis() + "" + rand.nextInt(500);
        Log.d("uniqueId", uniqueId);
        WMPreference.INSTANCE.set_UniqueId(mContext, uniqueId);
    }

    private void goToRegistrationActivity() {
        Thread background = new Thread() {
            @Override
            public void run() {
                try {
                    sleep(3 * 1000);
                    Intent i = new Intent(getBaseContext(), WelcomeActivity.class);
                    startActivity(i);
                    finish();
                } catch (Exception e) {
                }
            }
        };
        background.start();
    }

}

File: app/src/main/java/com/app/washmania/SplashActivity.kt
package com.app.washmania

import android.content.Context
import android.content.Intent
import android.os.Handler
import android.util.Log
import com.app.washmania.others.BaseActivity
import com.app.washmania.others.Preference.isgenerateUniqueKey
import com.app.washmania.others.Preference.setUniqueKey
import com.app.washmania.others.Preference.set_UniqueId
import java.util.*

class SplashActivity : BaseActivity() {

    lateinit var handler: Handler
    val waitingTime: Long = 3000
    var appdata: Boolean = true
    var context: Context? = null

    companion object {
        val CLASS_NAME = "SplashActivity"
    }


    override fun initResources() {
    }

    override fun initListeners() {
        context = this
        if (!isgenerateUniqueKey(context!!)) {
            generateUniqueId()
            setUniqueKey(context!!, true)
        }

        if (appdata) {
            movetopasscodescreen()
        } else {
            waitScreen()
        }
    }

    override fun getLayout(): Int {
        return R.layout.activity_splash
    }

    private fun waitScreen() {
        handler = Handler()
        handler.postDelayed(
            object : Runnable {
                override fun run() {
                    startActivity(Intent(this@SplashActivity, DashboardActivity::class.java))
                    finish()
                }
            }, waitingTime
        )
    }

    private fun movetopasscodescreen() {
        handler = Handler()
        handler.postDelayed(
            object : Runnable {
                override fun run() {
                    startActivity(Intent(this@SplashActivity, WelcomeActivity::class.java))
                    finish()
                }
            }, waitingTime
        )
    }

    private fun generateUniqueId() {
        val rand = Random()
        val uniqueId = System.currentTimeMillis().toString() + "" + rand.nextInt(500)
        Log.d("uniqueId", uniqueId)
        set_UniqueId(context!!, uniqueId)
    }
}
